proxy.go

```diff
@@ -103,6 +106,11 @@ func (p *Proxy) ServeConn(tenant net.Conn) error {
 		handShake.log.WithError(err).Error("close error")
 	}
 
+	// if connection was closed and it's the main channel, call the closeCallback
+	if handShake.channelType == red.ChannelMain && p.closeCallback != nil {
+		p.closeCallback(handShake.destination)
```

Comment:
Please add a `nil` check for when the `p.closeCallback` is not defined.

If defined:
Please log the execution on the callback function.
If an error occurs during the callback, this should be logged instead.

Reply:
done